sis_provisioner/account_managers/author_loader.py

```diff
@@ -0,0 +1,74 @@
+"""
+This class will load all the Bridge authors, check against gws author group
+1. Delete former authors
+2. Add new authors
+"""
+
+import logging
+import traceback
+from sis_provisioner.dao.gws import get_bridge_authors
+from sis_provisioner.dao.pws import get_person
+from sis_provisioner.dao.uw_account import save_uw_account
+from sis_provisioner.account_managers.gws_bridge import GwsBridgeLoader
+
+logger = logging.getLogger(__name__)
+
+
+class AuthorChecker(GwsBridgeLoader):
+
+    def __init__(self, worker, clogger=logger):
+        super(AuthorChecker, self).__init__(worker, clogger)
+
+    def fetch_users(self):
+        self.cur_author_set = get_bridge_authors()
+        self.data_source = "Bridge authors"
+        return self.get_bridge().get_all_authors()
+
+    def process_users(self):
+        existing_bri_authors = set()
+        # 1. remove former authors
+        for bri_acc in self.get_users_to_process():
+            uwnetid = bri_acc.netid
+            if uwnetid not in self.cur_author_set:
+                self.remove_author_role(bri_acc)
+            else:
+                existing_bri_authors.add(uwnetid)
+
+        # 2. add new authors
+        for netid in list(self.cur_author_set):
+            if self.in_uw_groups(netid) is False:
+                continue
+            if netid not in existing_bri_authors:
+                self.add_author_role(netid)
```

Comment:
Have you thought about using list addition and subtraction to get the diff here?

Reply:
Ah,  thank you!